File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        
        Set<Integer> numbers = new HashSet();
        for(int num: nums){
            if(!(numbers.contains(num))) {
                numbers.add(num);
            }else {
                return true;
            }
        }
        return false;
        
    }
}

File: Data Structures & Algorithms/search-insert-position/submission-1.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l =0 
        r = len(nums) - 1

        while l <= r:
            m = (r + l ) // 2

            if nums[m] <= target:
                l += 1
                if nums[m] == target:
                    return m
            else:
                r -= 1
        return l
